packages/arb-token-bridge-ui/tests/e2e/specs/depositERC20.cy.ts

```diff
@@ -0,0 +1,149 @@
+      cy.findByRole('button', { name: /To: Arbitrum Goerli/i }).should(
+        'be.visible'
+      )
+      cy.findByRole('button', { name: 'Select Token' })
+        .should('be.visible')
+        .should('have.text', 'ETH')
+    })
+
+    it('should should add ERC20 token correctly', () => {
+      // Click on the ETH dropdown (Select token button)
+      cy.findByRole('button', { name: 'Select Token' })
+        .should('be.visible')
+        .should('have.text', 'ETH')
+        .click({ scrollBehavior: false })
+
+      // open the Select Token popup
+      cy.findByPlaceholderText(/Search by token name/i)
+        .should('be.visible')
+        .type(ERC20TokenAddressL1, { scrollBehavior: false })
+        .then(() => {
+          // Click on the Add new token button
+
+          cy.findByRole('button', { name: 'Add New Token' })
+            .should('be.visible')
+            .click({ scrollBehavior: false })
+
+          // Select the LINK token
+          cy.findByText('ChainLink Token').click({ scrollBehavior: false })
+
+          // LINK token should be selected now and popup should be closed after selection
+          cy.findByRole('button', { name: 'Select Token' })
+            .should('be.visible')
+            .should('have.text', 'LINK')
+        })
+    })
+
+    it('should show ERC20 balance correctly', () => {
+      cy.findByText(`Balance: ${l1ERC20bal} LINK`).should('be.visible')
+    })
+
+    context("bridge amount is lower than user's L1 ERC20 balance value", () => {
+      it('should show summary', () => {
+        cy.findByPlaceholderText('Enter amount')
+          .type('0.0001', { scrollBehavior: false })
+          .then(() => {
+            cy.findByText('You’re moving')
+              .siblings()
+              .last()
+              .contains('0.0001 LINK')
+              .should('be.visible')
+            cy.findByText('You’ll pay in gas fees')
+              .siblings()
+              .last()
+              .contains(zeroToLessThanOneETH)
+              .should('be.visible')
+            cy.findByText('L1 gas')
+              .parent()
+              .siblings()
+              .last()
+              .contains(zeroToLessThanOneETH)
+              .should('be.visible')
+            cy.findByText('L2 gas')
+              .parent()
+              .siblings()
+              .last()
+              .contains(zeroToLessThanOneETH)
+              .should('be.visible')
+          })
+      })
+
+      it('should deposit successfully', () => {
+        cy.findByRole('button', {
+          name: 'Move funds to Arbitrum Goerli'
+        })
+          .click({ scrollBehavior: false })
+          .then(() => {
+            cy.confirmMetamaskTransaction().then(() => {
+              cy.findByText(/Moving 0.0001 LINK to Arbitrum Goerli.../i).should(
+                'be.visible'
+              )
+            })
+          })
+      })
+    })
+
+    // TODO => test for bridge amount higher than user's L1 ERC20 balance
+  })
+
```

Comment:
We should add a TODO test case which includes a token spend approval request when the user deposits that token for the first time.
Probably need to set approval amount to 0 after the test?

Reply:
Done - added a todo.